File: java/BubbleSort/src/bubblesort/BubbleSort.java
package bubblesort;

import java.util.ArrayList;


public class BubbleSort {

    public static void sort(ArrayList<Integer> array){
        int size  = array.size();
        
        for(int i = 0; i < size-1; i++){
            for(int j = 0; j < size-1; j++){
                if(array.get(j) > array.get(j+1)){
                    int aux = array.get(j);
                    array.set(j, array.get(j+1));
                    array.set(j+1, aux);
                }
            }    
        }
    }
    
        public static void print(ArrayList<Integer> array){
        
        for(int i = 0; i < array.size(); i++){
            System.out.print(array.get(i) + " ");
        }
        System.out.println();
       
        }
    
    public static void main(String[] args) {
         ArrayList<Integer> array = new ArrayList<>();
        array.add(5);
        array.add(65);
        array.add(20);
        array.add(18);
        array.add(1);
        array.add(63);
        array.add(2);
        array.add(32);

        print(array);
        sort(array);
        print(array);
    }
    
}

File: go/src/sort/bubbleSort.go
package sort

func bubbleSort(arr []int) {
	for i := 0; i < len(arr)-1; i++ {
		c := false
		for j := len(arr) - 1; j > i; j-- {
			if arr[j] < a[j-1] {
				x := arr[j]
				arr[j] = a[j-1]
				arr[j-1] = x
				c = true
			}
		}

		if !c {
			break
		}
	}
}
